### ISSUE TITLE
goBack() taking me all the way to first entry instead of where I want to be

### ISSUE THREAD

**Issue by kkotwal94** [rackt/react-router#2625]
I'm creating a app where I Login/Logout have a home, about, profile etc. Its using react router as well, however when I log in I want the app to transition to the last route it was at, it works sometimes for example if I visit about, then login and log in, it takes me back to about, however if i visit the app root '/' and then go to login, and login, it takes me back to the first page I was ever on (which would be the default page when you open your browser or tab). I wrote it like this, and i think the way I may have done it is very naive. I did it in the render method which is called alot. I think it has sometime do with that but am not entirely sure. Was hoping for some advice, code:

```
constructor(props, context) {
  	super(props, context);
  	this.state = UserStore.getState();
  	this.context = context;
  }

render() {
    
  	console.log(this.state.user.get('authenticated'));
  	if(this.state.user.get('authenticated')){
  		this.context.history.goBack();
  	}

     return (
    //stuff
    )
```

basically just saying, if user is authenticated finally, take me back one selection in history, tried history.go(-1) as well, got the same result.

**Comment by taion** [comment]
Thanks for your question!

We want to make sure that the GitHub issue tracker remains the best place to track bug reports and feature requests that affect the development of React Router.

Questions like yours deserve a purpose-built Q&A forum. Would you like to post this question to Stack Overflow with the tag #react-router? https://stackoverflow.com/questions/ask?tags=react-router.

We also have an active and helpful React Router community on Reactiflux, which is a great place to get fast help with React Router and with the rest of the React ecosystem. You can join at https://discord.gg/0ZcbPKXt5bYaNQ46.

**Comment by kkotwal94** [comment]
Yea, apologies, i posted it here:

https://stackoverflow.com/questions/34024567/goback-taking-me-all-the-way-to-first-entry-instead-of-where-i-want-to-be
